eureka-client/src/main/java/com/netflix/discovery/util/DeserializerStringCache.java

```diff
@@ -0,0 +1,77 @@
+package com.netflix.discovery.util;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.ObjectReader;
+import com.fasterxml.jackson.databind.cfg.ContextAttributes;
+
+/**
+ * A non-locking alternative to {@link String#intern()} and {@link StringCache} that works with DeserializationContext.  Definitely
+ * NOT thread-safe.
```

Comment:
the raison d'etre of DeserializerStringCache is to cache in a single-threaded context and thereby avoid the expense of locking.  I'll update the class-level documentation to call this out more clearly.